### PR TITLE
Add option to disable encoding [ch58602]

### PR DESCRIPTION
**Pull request by elenatorro**
This PR adds:

- Adds the new `encode_data`parameter to the Layer
- Captures the error from `pako` when there's been a problem decoding the data
- Shows an error that explains how to add set the new `encode_data` parameter in the Layer, so the user can easily copy-paste it

To Do:
- [x] Add to reference properly

encode

### CODE REVIEW

File: cartoframes/assets/src/map/SourceFactory.js [hunk 1]
```diff
@@ -30,6 +30,18 @@ function MVT(layer) {
   return new carto.source.MVT(layer.data.file, JSON.parse(layer.data.metadata));
 }
 
-function _decodeJSONData(b64Data) {
-  return JSON.parse(pako.inflate(atob(b64Data), { to: 'string' }));
+function _decodeJSONData(data, encodeData) {
+  try {
+    if (encodeData) {
+      const decodedJSON = pako.inflate(atob(data), { to: 'string' });
```
Comment (Jesus89) [inline cartoframes/assets/src/map/SourceFactory.js]: Should we do the `try` here?

File: cartoframes/utils/utils.py [hunk 2]
```diff
@@ -272,11 +272,15 @@ def default(self, o):
         return super(CustomJSONEncoder, self).default(o)
 
 
-def encode_geodataframe(data):
+def parse_local_data(data, encode=True):
```
Comment (Jesus89) [inline cartoframes/utils/utils.py]: I would prefer a more descriptive name :thinking:

File: cartoframes/assets/src/map/SourceFactory.js (hunk 1 above)
Comment (elenatorro) [reply cartoframes/assets/src/map/SourceFactory.js]: 👍

File: cartoframes/utils/utils.py (hunk 2 above)
Comment (elenatorro) [reply cartoframes/utils/utils.py]: I agree! What about `enable_encoding`? Any ideas?

Comment (Jesus89) [reply cartoframes/utils/utils.py]: Since we have `get_geodataframe_bounds`, something like `get_geodataframe_data` sounds good to me.

**Pull request merged**